src/routes/backupTarget/CreateBackupTargetModal.js

```diff
@@ -0,0 +1,115 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { Form, Input, Icon, InputNumber, Tooltip } from 'antd'
+import { ModalBlur } from '../../components'
+const FormItem = Form.Item
+
+const formItemLayout = {
+  labelCol: {
+    span: 6,
+  },
+  wrapperCol: {
+    span: 15,
+  },
+}
+
+const modal = ({
+  item,
+  allBackupTargetsName,
+  visible,
+  onCancel,
+  onOk,
+  form: {
+    getFieldDecorator,
+    validateFields,
+    getFieldsValue,
+    getFieldValue,
+  },
+}) => {
+  function handleOk() {
+    validateFields((errors) => {
+      if (errors) {
+        return
+      }
+      const data = {
+        ...getFieldsValue(),
+        credentialSecret: getFieldValue('credentialSecret')?.trim() || '',
+        backupTargetURL: getFieldValue('backupTargetURL')?.trim() || '',
+        pollInterval: getFieldValue('pollInterval')?.toString() || item.pollInterval.toString(), // pollInterval is a string type
+      }
+      onOk(data)
+    })
+  }
+
+  const modalOpts = {
+    title: 'Create Backup Target',
+    visible,
+    onCancel,
+    width: 700,
+    onOk: handleOk,
+  }
+
+  return (
+    <ModalBlur {...modalOpts}>
+      <Form layout="horizontal">
+        <FormItem label="Name" hasFeedback {...formItemLayout}>
+          {getFieldDecorator('name', {
+            initialValue: item.name,
+            rules: [
+              {
+                required: true,
+                message: 'Backup target name is required',
+              },
+              {
+                validator: (_rule, value, callback) => {
+                  if (allBackupTargetsName.includes(value)) {
+                    callback(`"${value}" is duplicated to existing backup target name`)
+                  } else {
+                    callback()
+                  }
+                },
+              },
+            ],
+          })(<Input />)}
+        </FormItem>
+        <FormItem label="URL" hasFeedback {...formItemLayout}>
```

Comment:
Should `URL` be required?

Reply:
It's required to render URL field title.

<img width="754" alt="Screenshot 2024-12-18 at 10 34 16 AM" src="https://github.com/user-attachments/assets/8d09506a-f1bb-4715-ad40-7323fe1069ec" />

Reply:
I mean should it be a required input field?